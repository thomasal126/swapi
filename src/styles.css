@import url("https://fonts.googleapis.com/css2?family=Poller+One&display=swap");

.App {
  font-family: "Poller One", sans-serif;
  color: white;
}
