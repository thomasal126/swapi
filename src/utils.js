export const processData = ({ gender, minHeight }) => (data) => {
  return data
    .filter((person) => person.gender === gender && person.height > minHeight)
    .sort((a, b) => (a.mass === b.mass ? b.height - a.height : b.mass - a.mass))
    .map(({ name, mass, height }) => ({
      name: name,
      mass: mass,
      height: height
    }));
};
