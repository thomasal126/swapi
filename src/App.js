import React, { useState, useEffect } from "react";
import axios from "axios";
import "./styles.css";
import { processData } from "./utils";
import Person from "./components/Person";
import background from "./image/background.jpg";

export default function App() {
  // const people1 = usePeopleData(filterPeopleParams); // Un-comment on Task #3
  const payloadURL = "https://swapi.dev/api/people/";
  const [people, setPeople] = useState([]);
  const [loading, setLoading] = useState(true);
  const [find, setFind] = useState("");

  const usePeopleData = (filterParams) => {
    useEffect(() => {
      async function fetchPeople(api) {
        try {
          let res = await axios.get(api);

          setPeople(res.data.results);
          setLoading(false);
        } catch (err) {
          console.log(err);
        }
      }
      fetchPeople(payloadURL);
    }, []);

    const display = processData(filterParams)(people);
    return display;
  };

  const filterPeopleParams = { gender: "male", minHeight: 130 };

  const person = usePeopleData(filterPeopleParams);

  const handleSearch = (e) => {
    setFind(e.target.value);
  };

  const readyToDisplay = person.filter((elem) =>
    elem.name.toLowerCase().includes(find.toLowerCase())
  );

  return (
    <div
      className="App"
      style={{
        backgroundImage: `url(${background})`,
        backgroundSize: "cover",
        height: "100vh"
      }}
    >
      <div>
        <label>
          Mission name:{" "}
          <input
            onChange={handleSearch}
            value={find}
            placeholder="Search Hero"
          />
        </label>
      </div>
      <h3>Heroes</h3>
      {loading ? (
        <p>Loading...</p>
      ) : (
        <div>
          <div className="mx-auto container-fluid row border border-danger">
            <h5 className="col-sm-4">Name</h5>
            <h5 className="col-sm-4">Mass</h5>
            <h5 className="col-sm-4">Weight</h5>
          </div>
          {readyToDisplay.map((person, index) => (
            <Person key={index} {...person} />
          ))}
        </div>
      )}
    </div>
  );
}
