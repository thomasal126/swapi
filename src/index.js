import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import "bootstrap/dist/css/bootstrap.css";

const rootElement = document.getElementById("app");
ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  rootElement
);
