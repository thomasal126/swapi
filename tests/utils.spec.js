import { processData } from "../src/utils";
import payload from "./__fixtures__/payload.json";

describe(`processData`, () => {
  test.each`
    gender      | minHeight
    ${"male"}   | ${130}
    ${"male"}   | ${180}
    ${"female"} | ${120}
    ${"female"} | ${160}
  `(
    `consolidates the data correctly for gender $gender and minHeight $minHeight`,
    ({ gender, minHeight }) => {
      expect(processData({ gender, minHeight })(payload)).toMatchSnapshot();
    }
  );
});
