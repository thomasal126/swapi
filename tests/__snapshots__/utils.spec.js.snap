// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processData consolidates the data correctly for gender female and minHeight 120 1`] = `
Array [
  Object {
    "height": "165",
    "mass": "75",
    "name": "Beru Whitesun lars",
  },
  Object {
    "height": "150",
    "mass": "49",
    "name": "Leia Organa",
  },
]
`;

exports[`processData consolidates the data correctly for gender female and minHeight 160 1`] = `
Array [
  Object {
    "height": "165",
    "mass": "75",
    "name": "Beru Whitesun lars",
  },
]
`;

exports[`processData consolidates the data correctly for gender male and minHeight 130 1`] = `
Array [
  Object {
    "height": "202",
    "mass": "136",
    "name": "Darth Vader",
  },
  Object {
    "height": "178",
    "mass": "120",
    "name": "Owen Lars",
  },
  Object {
    "height": "183",
    "mass": "84",
    "name": "Biggs Darklighter",
  },
  Object {
    "height": "182",
    "mass": "77",
    "name": "Obi-Wan Kenobi",
  },
  Object {
    "height": "172",
    "mass": "77",
    "name": "Luke Skywalker",
  },
]
`;

exports[`processData consolidates the data correctly for gender male and minHeight 180 1`] = `
Array [
  Object {
    "height": "202",
    "mass": "136",
    "name": "Darth Vader",
  },
  Object {
    "height": "183",
    "mass": "84",
    "name": "Biggs Darklighter",
  },
  Object {
    "height": "182",
    "mass": "77",
    "name": "Obi-Wan Kenobi",
  },
]
`;
